site/_posts/2020-09-03-oidc-auth-api-usage.md

```diff
@@ -78,15 +83,39 @@ oidc_client_secret="12345678-1234-1234-1234-01234567abcd"
 # OIDCClientID from the OpenID-Connect configuration.
 oidc_client_id="my-oidc-client"
 
-# OIDCOauthIntrospectionEndpoint from the OpenID-Connect configuration
-oidc_auth_introspection_endpoint="https://keycloak.example.com/auth/realms/miq/protocol/openid-connect/token/introspect"
-
-# OIDCProviderTokenEndpoint from the OpenID-Connect configurationt
-# Sometimes OIDCOauthIntrospectionEndpoint without the trailing "introspect"
-oidc_provider_token_endpoint="https://keycloak.example.com/auth/realms/miq/protocol/openid-connect/token/"
 ```
 
-## Example: Using a JWT Token
+Each OpenID-Connect Provider defines its own endpoints for token generation and introspection. Consequently
+the value of these variables depend on the OpenID-Connect Provider being used.
+
+**Example for IBM's Identity Access Management (IAM):**
+
+> Given this configuration:
+> ```bash
+> OIDCOAuthIntrospectionEndpoint https://cp-console.apps.iam.example.com/idprovider/v1/auth/introspect
+> OIDCProviderTokenEndpoint      https://cp-console.apps.iam.example.com/idprovider/v1/auth/token
+> ```
+>
+> Then **oidc_auth_introspection_endpoint** and **oidc_provider_token_endpoint** will be:
+> ```bash
+> oidc_auth_introspection_endpoint="https://cp-console.apps.iam.example.com/idprovider/v1/auth/introspect"
+> oidc_provider_token_endpoint="https://cp-console.apps.iam.example.com/idprovider/v1/auth/identitytoken"
+> ```
+
+**Example for Keycloak:**
+
+> Given this configuration:
+> ```bash
+> OIDCOAuthIntrospectionEndpoint https://keycloak.example.com/auth/realms/miq/protocol/openid-connect/token/introspect 
+> ```
```

Comment:
how come there is no OIDCProviderTokenEndpoint here ?

Reply:
@abellotti 

Because keycloak supports OIDCProviderMetadataURL.

see: https://github.com/zmartzone/mod_auth_openidc/blob/master/auth_openidc.conf#L63

Reply:
@abellotti Did you just say "oh yeah, now I remeber" ? 😄

Reply:
oh that !! 😄